File: src/main/java/shopping/auth/JwtProvider.java
package shopping.auth;

import io.jsonwebtoken.Jwts;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;
import io.jsonwebtoken.security.Keys;
import java.security.Key;
import java.time.Duration;
import java.util.Date;

@Component
public class JwtProvider {

    private final Key key;
    private static final long TOKEN_EXPIRATION_TIME = Duration.ofHours(1).toMillis();

    public JwtProvider(@Value("${jwt.secret}") String secret) {
        this.key = Keys.hmacShaKeyFor(secret.getBytes());
    }

    public String createToken(String email) {

        Date now = new Date();
        Date expiry = new Date(now.getTime() + TOKEN_EXPIRATION_TIME);

        return Jwts.builder()
                .setSubject(email)
                .setIssuedAt(now)
                .setExpiration(expiry)
                .signWith(key)
                .compact();
    }
}

File: src/main/kotlin/shopping/common/util/JwtProvider.kt
package shopping.common.util

import io.jsonwebtoken.Jwts
import io.jsonwebtoken.security.Keys
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component
import java.util.Date

const val ONE_WEEK_MILLISECOND: Long = 1000 * 60 * 60 * 24 * 7

@Component
class JwtProvider(
    @Value("\${jwt.secret-key}")
    private val secretKey: String,
    @Value("\${jwt.expiration-millis-second}")
    private val expirationMillisSecond: Long = ONE_WEEK_MILLISECOND,
) {
    private val signingKey = Keys.hmacShaKeyFor(secretKey.toByteArray())

    fun createToken(subject: String): String {
        val now = Date()
        val expireAt = Date(now.time + expirationMillisSecond)

        return Jwts
            .builder()
            .subject(subject)
            .issuedAt(now)
            .expiration(expireAt)
            .signWith(signingKey)
            .compact()
    }
}
